dotnetv3/dynamodb/low-level-api/LowLevelTableExample/LowLevelTableExample.cs

```diff
@@ -4,222 +4,210 @@
+        Console.WriteLine($"Provision Throughput (writes/sec): " +
+                          $"{table.ProvisionedThroughput.WriteCapacityUnits}");
+    }
+    // snippet-end:[dynamodb.dotnetv3.DescribeTableExample]
 
-            return true;
-        }
-        // snippet-end:[dynamodb.dotnetv3.DescribeTableExample]
+    // snippet-start:[dynamodb.dotnetv3.UpdateExampleTable]
+    private static async Task UpdateExampleTable()
+    {
+        Console.WriteLine("\n*** Updating table ***");
 
-        public static async Task<bool> UpdateExampleTable(AmazonDynamoDBClient client)
+        await Client.UpdateTableAsync(new UpdateTableRequest
         {
-            Console.WriteLine("\n*** Updating table ***");
-            var request = new UpdateTableRequest()
+            TableName = ExampleTableName,
+            ProvisionedThroughput = new ProvisionedThroughput
             {
-                TableName = _tableName,
-                ProvisionedThroughput = new ProvisionedThroughput()
-                {
-                    ReadCapacityUnits = 6,
-                    WriteCapacityUnits = 7
-                }
-            };
-
-            await client.UpdateTableAsync(request);
+                ReadCapacityUnits = 6,
+                WriteCapacityUnits = 7
+            }
+        });
 
-            WaitUntilTableReady(client);
+        await WaitUntilTableReady(ExampleTableName);
+    }
+    // snippet-end:[dynamodb.dotnetv3.UpdateExampleTable]
 
-            return true;
-        }
+    // snippet-start:[dynamodb.dotnetv3.DeleteExampleTable]
+    private static async Task DeleteExampleTable()
+    {
+        Console.WriteLine("\n*** Deleting table ***");
 
-        public static async Task<bool> DeleteExampleTable(AmazonDynamoDBClient client)
+        try
         {
-            Console.WriteLine("\n*** Deleting table ***");
-            var request = new DeleteTableRequest
+            await Client.DeleteTableAsync(new DeleteTableRequest
             {
-                TableName = _tableName
-            };
-
-            await client.DeleteTableAsync(request);
+                TableName = ExampleTableName
+            });
 
             Console.WriteLine("Table is being deleted...");
-
-            return true;
         }
+        catch (ResourceNotFoundException)
+        {
+            // something went wrong during CreateTable
+        }
+    }
+    // snippet-end:[dynamodb.dotnetv3.DeleteExampleTable]
 
-        private static async void WaitUntilTableReady(AmazonDynamoDBClient client)
+    // snippet-start:[dynamodb.dotnetv3.WaitUntilTableReady]
+    private static async Task WaitUntilTableReady(string tableName)
+    {
+        string? status = null;
+        // Let us wait until table is created. Call DescribeTable.
+        do
         {
-            string status;
-            // Let us wait until table is created. Call DescribeTable.
-            do
+            Thread.Sleep(5000); // Wait 5 seconds.
```

Comment:
Are implicit waits the only wait to handle this? I'd be worried this could fail for a user a certain percentage of the time.

Reply:
I think it would be better to get the table description until the table exists.

Reply:
I see. I missed the context of this being within a 'WaitUntilTableReady' function already. My mistake!